### ISSUE TITLE
Add description in the strict-blocking page

### ISSUE THREAD

**Issue by Yuki2718** [uBlockOrigin/uBlock-issues#2476]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

I found Brave's strict-blocking page (only on Aggressive mode) has a characteristic that uBO should probably follow.

![brave-warn](https://user-images.githubusercontent.com/58900598/216640023-e145fa23-0bb6-4720-8b4f-7feebedae30d.png)

Although it's not necessarily tracking, they explain what had happened. There are only three patterns to trigger strict-blocking:
1. malicious site
2. click-through tracker or advertiser's site
3. false positive

Let's ignore the last case and advertiser's sites as are rare. I believe uBO can and should add explanation of what happened, which is most likely to be either malicious site or tracker - then let user to decide which it is. This partly solves https://github.com/uBlockOrigin/uBlock-issues/issues/1195 and is much easier to implement. The current blocking page consists of only facts is too unfriendly to average Joe to whom "EasyList" is just mumbo jumbo. After I started Twitter I occasinally see user get confused by the blocking page; e.g. `https://twitter.com/_1__1_/status/1615997416976834565`, and this has long been my concern. This suggestion should be a step ahead for uBO to rally be install-and-forget.

### A specific URL where the issue occurs.

https://www.profitablegatetocontent.com/e51xmfb9?key=6cf0bf53774e52ec9e3ca94803f48b06

### Steps to Reproduce

1. Visit the page and strict-blocking will be triggered by EL
2. See how much it explains to average Joe

### Expected behavior

Written

### Actual behavior

Written

### uBO version

1.46.0

### Browser name and version

Firefox 109.0.1

### Operating System and version

Windows 10

**Comment by garry-ut99** [comment]
Duplicate of https://github.com/gorhill/uBlock/issues/1445 which was declined.

**Comment by Yuki2718** [comment]
> Duplicate of 

How it is?
> uBO can and should add explanation of what happened, which is most likely to be either malicious site or tracker - then let user to decide which it is.

The proposal here is that uBO should add an explanation universal to all the strict-blocking pages regardless of the list or rule causing it.

**Comment by gothic-bum** [comment]
> The proposal here is that uBO should add an explanation universal to all the strict-blocking pages regardless of the list or rule causing it.

How can there be a universal explanation when uBO has no way of determining for certain why a page is strict blocked? 
  
For example:  
If a filter list - or a custom filter added by a user - contains `||example-site.com^`, how could uBO possibly determine the reason(s) why it was included? 

It could have been added for a variety of reasons, and I personally have hundreds of custom strict block filters that uBO has no way of knowing why I added them or what category they are (e.g. tracker, malicious site, advertiser, annoyance).

So an "explanation" would be nothing more than a guess and most likely incorrect - which could be actually be **more** confusing to the user than what is shown with the current message on the strict block page.

**Comment by Yuki2718** [comment]
We don't need to care about non-default settings, those who configure settings should have better understanding. For default lists, only covering bad sites and tracker is 99% accurate. The more context of this is that it appears user tend to not be aware of the possibility of click-through tracking.

**Comment by gothic-bum** [comment]
> For default lists, only covering bad sites and tracker is 99% accurate. 

EasyList is a default list. It currently has 19857 entries that are strict blocked and are **not** categorized as trackers or bad sites.
  
Peter Lowe's list is also a default list. It currently has 2614 entries (71% out of 3667) that are strict blocked and are **not** categorized as trackers.

**Comment by Yuki2718** [comment]
I mean those actually causing strict-blocking in real world is mostly either one. The example here `ww.profitablegatetocontent.com` is an adserver but is perfectly fine to be classified to both as it tracks and also leads to bad sites with very high certainty.

**Comment by Yuki2718** [comment]
Anyway my point is similar to those issues @garry-ut99 linked, the current page is way too unfrinedly and is *actually* confusing user, I'm not talking about a theoretical scenario.

**Comment by krystian3w** [comment]
You can write database to dynamic classify as `ad server`, `tracking server`, `malware server`.

**Comment by gorhill** [comment]
The example you provide is a Tweet of a user who observes that uBO blocked the navigation. He isn't saying he is confused. uBO already clearly state on that page that the page was blocked as a result of a filter and offers course of action to the user. What's the point of telling them there is no worry and just click proceed when the whole goal is to make them aware and let them decide for themselves what is best?

**Comment by krystian3w** [comment]
Apparently so, but when my mother browses through the shops she is constantly annoyed at the strict blockers and I usually have to unblock or explain to her that she is opening a fake online shop.

**Comment by Yuki2718** [comment]
I was forced to link that tweet as another one I tried to example was already deleted by the author, but the context of the tweet is he's applying an online seminer and there's no other choice than proceeding. If I had a fault, that would be not explained it's possible to use the loope icon to show a link without tracking.

**Comment by gorhill** [comment]
I really think a generic message downplaying what may be occurring is not the way to go if we want to provide more information about what is going on, consider:

![Screenshot from 2023-02-04 09-11-17](https://user-images.githubusercontent.com/585534/216772291-f003e9c0-4820-4992-8283-ad69fef06d95.png)

**Comment by Yuki2718** [comment]
A question is how many ppl understand, or even can read (as written in English) Online Malicious URL Blocklist.

**Comment by gorhill** [comment]
Well maybe a more descriptive message depending on the group in which a list is categorize would be the way, but it would never be as specific as making the assumption it's a cookie tracking instance.

**Comment by Yuki2718** [comment]
That was kinda declinded as @garry-ut99 linked, but happy if you reconsider.

**Comment by gorhill** [comment]
By the way, when I proceeded with you example link, `https://www.profitablegatetocontent.com/e51xmfb9?key=6cf0bf53774e52ec9e3ca94803f48b06`, I hit again another strict-blocked page as a result of the filter `/?pl=*&sm=$document` in _uBlock filters – Badware risks_, and then again with another filter in _uBlock filters – Badware risks_, and then again _uBlock filters – Badware risks_. I stopped clicking _Proceed_ at this point. This shows a generic message about cookie definitely is not the way to go.

**Comment by Yuki2718** [comment]
The link 99% leads to bad sites. So I'm saying in "A or B" manner. I commented to Brave that they should include the notion of bad sites.

**Comment by krystian3w** [comment]
Perhaps scam blockers are added too quickly based on URL parameters.

**Comment by gorhill** [comment]
> That was kinda declinded as @garry-ut99 linked

Yes, and today I read again the issue and I think it was reasonable request. This was years ago and I admittedly go through period where I feel the burden of this project more than usual and in this case I will more eagerly dismiss issues.

**Comment by garry-ut99** [comment]
> https://github.com/uBlockOrigin/uBlock-issues/issues/2476#issuecomment-1416657934 : How it is?

A very slightly different solution to an identical problem still falls into the duplicate category. (and yes, I know the previous thread is old and locked and that devs sometimes change their minds after years, I only informed about the same issue already existing, nothing more)

> I think it was reasonable request.

Me too, small info won't hurt, only help, for majority of unadvanced users.

**Comment by Yuki2718** [comment]
> it's possible to use the loope icon to show a link without tracking.

BTW why it's hidden by default?

**Comment by garry-ut99** [comment]
It's not clear what do you mean by "it": just "a link without tracking." or the whole spoiler with all the parameters, in case of the latter, it might be:
- for aesthetic reasons, to not clutter the screen with technical guts
- to not bloat unexperienced users (which are the majority) by too much technical info they don't understand anyway
- after clicking loupe icon and the spoiler gets uncollapsed, the uncollapsed view is remembered and no need to click loupe icon anymore (until you collapse it again)

**Comment by Yuki2718** [comment]
Once the idea of trusted list comes true and if [once-declined](https://github.com/uBlockOrigin/uBlock-issues/issues/760#issuecomment-720080024) `$queryjump` was implemented in future, these can be solved without compromising privacy or user experience.

**Comment by Yuki2718** [comment]
To say the least, most user can't think of the possibility of tracking link:
```
https://twitter.com/Kota_pclive/status/1642870419408699394
https://twitter.com/metalch/status/1643951418884378624
```
and given most of those are by PL list and it's under Multipurpose, showing catogory won't help much. I wonder if there's a safe way to imply the blocking **may** be due to tracking link to user (with localizable way).

**Comment by gorhill** [comment]
I rather keep it simple if we are going to do something about this, adding per-filter message vastly complicate the filtering engine and completely sabotage all the efficiency work which went into it.

**Comment by iam-py-test** [comment]
> For example I block [scam](https://github.com/MasterKia/PersianBlocker/blob/d77387d9e0c51b57d67cb4d0411d53c98085e48b/PersianBlocker.txt#L3203) and [phishing/malware](https://github.com/MasterKia/PersianBlocker/blob/d77387d9e0c51b57d67cb4d0411d53c98085e48b/PersianBlocker.txt#L3360) in my list.

Maybe there could be a more general message (something similar to _This website may attempt to infect your device with malicious software, or steal your personal information or money_, although I'm sure someone else can come up with a better message).

**Comment by Elementari** [comment]
Mostly duplicate of [#3449](https://github.com/uBlockOrigin/uBlock-issues/issues/3449), where I pointed out a concrete scenario.

I really think a simple option to assign colors (to the button "proceed") depending on filter (i.e., by filter) and/or blocking reason (i.e., by page) but be of considerable use. 

In the case of assinging a color to a whole filter, it would still be about probabilities/heuristics. Just because a filter that's mostly about affiliate links might also help block malware, a dedicated malware filter is much more likely to be of concern and should rightfully be elevated optically as a warning/by using a red interface. Cf. Bayes' theorem, alpha/better error. I think this is a *serious* issue and possibly a likely cause of malware infection despite the use of uBlock.

Blue could be asigned to "neural" filters that block both malware and mere annoyances, green to those that block exclusively "benign" stuff (as defined by user; users might decide affiliate links are suffiently benign most of the time that tehy would want the green color for these, even if there's a remaining risk), and red to filters/reasons that are associated with high risk of malware. 

Like I said I'm not sure if there's a system that will prioritize malign blocking reasons over more benign ones, when both apply.

**Comment by krystian3w** [comment]
> I really think a simple option to assign colors

Just how to assign colors so that the computer recognizes it without commenting over the filter (so generate problem for EasyList/EasyPrivacy/Peter Lowe).
